docs/samples/python/ChatGPT_synth_data.ipynb

```diff
@@ -0,0 +1,422 @@
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b0f30aed",
+   "metadata": {},
+   "source": [
+    "## De-identify data using Presidio Analyzer and Anonymizer"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "d0dd7820",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "Hello, my name is <PERSON> and I live in <LOCATION>.\n",
+      "My credit card number is <CREDIT_CARD> and my crypto wallet id is <CRYPTO>.\n",
+      "\n",
+      "On <DATE_TIME> I visited <URL> and sent an email to <EMAIL_ADDRESS>,  from the IP <IP_ADDRESS>.\n",
+      "\n",
+      "My passport: <US_PASSPORT> and my phone number: <PHONE_NUMBER>.\n",
+      "\n",
+      "This is a valid International Bank Account Number: <IBAN_CODE> . Can you please check the status on bank account <US_BANK_NUMBER>?\n",
+      "\n",
+      "<PERSON>'s social security number is <US_SSN>.  Her driver license? it is <US_DRIVER_LICENSE>.\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "from presidio_analyzer import AnalyzerEngine\n",
+    "from presidio_anonymizer import AnonymizerEngine\n",
+    "\n",
+    "analyzer = AnalyzerEngine()\n",
+    "anonymizer = AnonymizerEngine()\n",
+    "\n",
+    "sample = \"\"\"\n",
+    "Hello, my name is David Johnson and I live in Maine.\n",
+    "My credit card number is 4095-2609-9393-4932 and my crypto wallet id is 16Yeky6GMjeNkAiNcBY7ZhrLoMSgg1BoyZ.\n",
+    "\n",
+    "On September 18 I visited microsoft.com and sent an email to test@presidio.site,  from the IP 192.168.0.1.\n",
+    "\n",
+    "My passport: 191280342 and my phone number: (212) 555-1234.\n",
+    "\n",
+    "This is a valid International Bank Account Number: IL150120690000003111111 . Can you please check the status on bank account 954567876544?\n",
+    "\n",
+    "Kate's social security number is 078-05-1126.  Her driver license? it is 1234567A.\n",
+    "\"\"\"\n",
+    "\n",
+    "results = analyzer.analyze(sample, language=\"en\")\n",
+    "anonymized = anonymizer.anonymize(text=sample, analyzer_results=results)\n",
+    "anonymized_text = anonymized.text\n",
+    "print(anonymized_text)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f77b0c0f",
+   "metadata": {},
+   "source": [
+    "## Create prompt (instructions + text to manipulate)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "d2a63fda",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_prompt(anonymized_text: str) -> str:\n",
+    "    \"\"\"\n",
+    "    Create the prompt with instructions to GPT-3.\n",
+    "    \n",
+    "    :param anonymized_text: Text with placeholders instead of PII values, e.g. My name is <PERSON>.\n",
+    "    \"\"\"\n",
+    "\n",
+    "    prompt = f\"\"\"\n",
+    "    You role is to create synthetic text based on de-identified text with placeholders instead of personally identifiable information.\n",
```

Comment:
Hey @omri374 

I changed the prompt a bit, removed unnecessary instructions and constructed 1 shot learning.
It seems it works fine.

prompt = f"""
Your role is to create synthetic text based on de-identified text with placeholders instead of personally identifiable information.
Replace the placeholders (e.g. <PERSON>, <SSN>, {{DATE}}, {{ip_address}}) with fake values.

Instructions:
* Use completely random numbers, so every digit is drawn between 0 and 9.
* Use realistic names that come from diverse genders, ethnicities and countries.
* If there are no placeholders, return the text as is and provide an answer.

input: How do I change the limit on my credit card {{credit_card_number}}?
output: How do I change the limit on my credit card 2539 3519 2345 1555?
input: {anonymized_text}
output:
"""

WDYT?

Reply:
Thanks! Let me check how it performs on the full data

Reply:
@erann1987 prompt integrated and tested. Please review.